src/c4v/scraper/crawler/crawlers/base_crawler.py

```diff
@@ -0,0 +1,130 @@
+
+        for sitemap in sitemaps:
+
+            # Get a sitemap from its url
+            sitemap_content = self.get_sitemap_from_url(sitemap)
+
+            # parse urls for the current sitemap
+            urls.extend(new_urls := self.parse_urls_from_sitemap(sitemap_content))
+
+            # process new urls if post process function exists
+            if post_process_data:
+                post_process_data(new_urls)
+
+        return urls
+
+    def get_sitemaps_from_index(self) -> List[str]:
+        """
+            Some sites may have its sitemap paginated in more sitemaps.
+            I such case, override this class. By default, it assumes 
+            the sitemap set is actually the starting one.
+        """
+        assert self.start_sitemap_url != None, "Start sitemap url not configured"
+
+        resp = requests.get(self.start_sitemap_url)
+        if resp.status_code != 200:
+            resp.raise_for_status()
+
+        return self.parse_sitemaps_urls_from_index(resp.text)
+
+    def parse_sitemaps_urls_from_index(self, sitemap_index: str) -> List[str]:
+        """
+            Get sitemap list from the content of an xml file with the filemap data
+            Parameters:
+                + sitemap_index : str = sitemap index xml content as string
+            Return:
+                List of urls listed by this index
+        """
+        soup = bs4.BeautifulSoup(sitemap_index, "xml")
+        urls = map(lambda l: l.get_text(), soup.find_all("loc"))
+        urls = filter(self.check_sitemap_url, urls)
+
+        return list(urls)
+
+    def get_sitemap_from_url(self, sitemap_url: str) -> str:
+        """
+            Get sitemap xml content from its corresponding url
+            Parameters:
+                + sitemap_url : str = sitemap url to retrieve
+            Return:
+                sitemap content if everything went ok
+        """
+        resp = requests.get(sitemap_url)
+        if resp.status_code != 200:
+            resp.raise_for_status()
+
+        return resp.text
+
+    def parse_urls_from_sitemap(self, sitemap: str) -> List[str]:
+        """
+            Given a sitemap body, parse site urls from it 
+            Parameters:
+                + sitemap : str = sitemap xml content as string
+            Return:
+                list of urls parsed for this sitemap
+        """
+        soup = bs4.BeautifulSoup(sitemap, "xml")
+        urls = map(lambda l: l.get_text(), soup.select("url > loc"))
+        urls = filter(self.check_page_url, urls)
+
+        return list(urls)
+
+    @staticmethod
+    def check_sitemap_url(url: str) -> bool:
+        """
+            Function to check if a given sitemap url
+            to another sitemap is a desired one
+            Parameters:
+                + url : str = url to check
+            return:
+                if this is a valid url
+        """
+        raise NotImplementedError("Implement this abstract method")
```

Comment:
If the plan is to override this method, should it be a class method instead?

Reply:
You mean something like:
```
@classmethod
def check_sitemap_url(cls, url: str)->bool
``` ?

why should it be better that way?